tests/integration/tests/conftest.py

```diff
@@ -44,10 +43,32 @@ def fixture_data(start_stop_cloudbeat, k8s, cloudbeat_agent):
     @param start_stop_cloudbeat: fixture to start and stop agent / cloudbeat
     @param k8s: Kubernetes wrapper object
     @param cloudbeat_agent: config object
-    @return: pods, nodes in cluster
+    @return: pods, nodes, leader_node in cluster
     """
     # pylint: disable=W0612,W0613
     pods = k8s.get_agent_pod_instances(agent_name=cloudbeat_agent.name,
                                        namespace=cloudbeat_agent.namespace)
     nodes = k8s.get_cluster_nodes()
-    return pods, nodes
+    leader_node = get_cluster_leader(k8s, cloudbeat_agent, pods)
```

Comment:
Why do you need leader information _before_ test case execution? You can just pass k8s object to the test function and retrieve the data inside the test case.